### DISCUSSION TITLE
Why is the useSuspenseQuery gcTime set to at least 1000, it is too long.

### DISCUSSION THREAD

**Discussion by xyhxx** [TanStack/query#8052] [Q&A]
I understand that set gcTime > 0 is to prevent infinite retries, but 1000 is a little too long.

For example, when I use useQuerySuspense results to useForm defaultValues, it is possible to use stale data.

Can we set a smaller gcTime like 10,100 here.

**Comment by TkDodo** [comment]
Can you elaborate when this is a problem please?

**Comment by xyhxx** [comment]
Yes.

https://stackblitz.com/edit/vitejs-vite-ujuh7q?file=src%2FApp.tsx

Is an example of similar.

I modify the user information in the modal and close the modal, if I open it again within 1 second, stale data will be used for defaultValues.

If I want to use the latest data I need to use useEffect and setValue, or removeQueries when unmount.

Or should I do it another way? Perhaps this is the wrong approach.

**Reply by TkDodo** [reply]
The issue we tried to fix when setting a "minimal" gcTime greater zero is because react re-renders the component that threw to an error boundary once before actually showing the error boundary. If we gc the query immediately (with 0, we have no timers), it will lead to another refetch because of that render.

Now it's questionable what "minimal" is. I think we've seen it sometimes fail with `gcTime: 10`. We could try `100` here instead of `1000` but it's hard to figure out if that is "enough". We chose 1second because it's the same we use for a minimal `staleTime`, too.

With react-hook-form, I would recommend the `reactive values` api that was made for these cases instead of `defaultValues`: https://react-hook-form.com/faqs#Howtoinitializeformvalues

**Reply by xyhxx** [reply]
Yes, that's one way. But this presents a new problem.

If the request takes 1 second to return data, the `input` will display the old data first and the new data after 1 second. And no `fallback `is displayed during this period. This is not a good UX.

[https://stackblitz.com/edit/vitejs-vite-fod2th?file=src%2FApp.tsx,src%2Fmain.tsx](url)

The best way to think of now is to call  `queryClient.removeQuestions` when unMount

**Reply by TkDodo** [reply]
you can check `isFetchedAterMount` to avoid displaying stale data
